plugins/firehydrant/src/components/ServiceDetailsCard/ServiceDetailsCard.tsx

```diff
@@ -0,0 +1,285 @@
+    color: '#FFF',
+    textTransform: 'none',
+    '&:hover': {
+      backgroundColor: '#614ab6',
+    },
+  },
+  buttonContainer: {
+    display: 'grid',
+    gridGap: '24px',
+    gridAutoFlow: 'column',
+    gridAutoColumns: 'min-content',
+  },
+  icon: {
+    color: '#f1642d',
+  },
+  link: {
+    textDecoration: 'underline',
+    fontSize: '16px',
+    lineHeight: '27px',
+    color: '#3b2492',
+    '&:hover, &:focus': {
+      fontWeight: '500',
+    },
+  },
+  linksContainer: {
+    borderBottom: '1px solid #d5d5d5',
+    padding: '10px 0px 10px 20px',
+    backgroundColor: '#f1edff',
+    marginBottom: '20px',
+  },
+  table: {
+    width: '100%',
+  },
+  warning: {
+    display: 'flex',
+    alignItems: 'center',
+    padding: '10px',
+    background: '#f1edff',
+    color: theme.palette.type === 'dark' ? '#3b2492' : '#fff',
+  },
+}));
+
+const ServiceWarning = ({ text }: { text: string }) => {
+  const classes = useStyles();
+  return (
+    <div className={classes.warning}>
+      <WarningIcon />
+      &nbsp;&nbsp;<span>{text}</span>
+    </div>
+  );
+};
+
+const ServiceAnalyticsView = ({
+  serviceId,
+  startDate,
+  endDate,
+}: {
+  serviceId: string;
+  startDate: moment.Moment;
+  endDate: moment.Moment;
+}) => {
+  const {
+    loading: analyticsLoading,
+    value: analyticsValue = {},
+    error: analyticsError,
+  } = useServiceAnalytics({
+    serviceId,
+    startDate: startDate.format('YYYY-MM-DD'),
+    endDate: endDate.format('YYYY-MM-DD'),
+  });
+
+  return (
+    <ServiceAnalytics
+      loading={analyticsLoading}
+      value={analyticsValue}
+      error={analyticsError}
+    />
+  );
+};
+
+export const ServiceDetailsCard = () => {
+  const { entity } = useEntity();
+  const classes = useStyles();
+  const [showServiceDetails, setShowServiceDetails] = useState(false);
+  const configApi = useApi(configApiRef);
+
+  const BASE_URL =
+    configApi.getOptionalString('firehydrant.baseUrl') ||
+    'https://app.firehydrant.io';
+
+  const startDate = moment().subtract(30, 'days').utc();
+  const endDate = moment().utc();
+
+  // The Backstage service name in FireHydrant is a unique formatted string
+  // that requires the entity's kind, name, and namespace.
+  const fireHydrantServiceName = `${entity?.kind}:${
+    entity?.metadata?.namespace ?? 'default'
+  }/${entity?.metadata?.name}`;
+
+  const { loading, value, error } = useServiceDetails({
+    serviceName: fireHydrantServiceName,
+  });
+
+  const activeIncidents: string[] = value?.service?.active_incidents ?? [];
+  const incidents: ServiceIncidentsResponse = value?.incidents ?? [];
+  const serviceId: string = value?.service?.id!;
+
+  useEffect(() => {
+    if (value?.service && Object.keys(value?.service).length > 0) {
+      setShowServiceDetails(true);
+    }
+  }, [value]);
+
+  if (loading) {
+    return <Progress />;
+  }
+
+  if (error) {
+    return (
+      <ServiceWarning text="There was an issue connecting to FireHydrant." />
```

Comment:
you may be interested in [ResponseErrorPanel](https://github.com/backstage/backstage/blob/dc05aef9e73d78382cb79ae3da0565c6874eb3d1/packages/core-components/src/components/ResponseErrorPanel/ResponseErrorPanel.tsx?plain=1) which nicely renders errors in a collapsible thing